src/query-builder/update-query-builder.ts

```diff
@@ -881,7 +881,8 @@ export class UpdateQueryBuilder<DB, UT extends keyof DB, TB extends keyof DB, O>
     return [
       new UpdateResult(
         // TODO: remove numUpdatedOrDeletedRows.
-        result.numAffectedRows ?? result.numUpdatedOrDeletedRows ?? BigInt(0)
+        result.numAffectedRows ?? result.numUpdatedOrDeletedRows ?? BigInt(0),
+        result.numChangedRows ?? BigInt(0)
```

Comment:
Since `numChangedRows` is an optional field, it should default to `undefined` instead of 0. 0 value would confuse the users that don't happen to read the comment that this is a MySQL-only field.

Reply:
I did this for consistency. `numAffectedRows` and `numChangedRows` both are optional in `QueryResult` but not in `UpdateResult`.


`driver/database-connection.ts`
```ts
export interface QueryResult<O> {
  readonly numAffectedRows?: bigint
  readonly numChangedRows?: bigint
}
```

`query-builder/update-result.ts`
```ts
export class UpdateResult {
  readonly #numUpdatedRows: bigint
  readonly #numChangedRows: bigint

  constructor(numUpdatedRows: bigint, numChangedRows: bigint) {
    this.#numUpdatedRows = numUpdatedRows
    this.#numChangedRows = numChangedRows
  }
}
```